app/controllers/collections_controller.rb

```diff
@@ -84,9 +84,10 @@ def push_collection
   end
 
   def download_all
-    binary_zip_data = ProformaService::ExportTasks.call(tasks: @collection.tasks)
-
+    binary_zip_data = ProformaService::ExportTasks.call(tasks: @collection.tasks, options: {version: params[:version]})
     send_data(binary_zip_data.string, type: 'application/zip', filename: "#{@collection.title}.zip", disposition: 'attachment')
+  rescue ProformaXML::PostGenerateValidationError => e
+    redirect_to :root, danger: JSON.parse(e.message).map {|msg| t("proforma_errors.#{msg}", default: msg) }.join('<br>')
```

Comment:
What about redirecting to `@collection`? I feel, this would increase usability.

Reply:
As discussed for tasks [see here](https://github.com/openHPI/codeharbor/pull/1550#discussion_r1715644539), we agreed to keep the redirect to `:root` since it occurs in a new tab.